### PR TITLE
allow domain names > HOST_NAME_MAX but < 255 in dhcp client

### PR DESCRIPTION
**Pull request by mischief**
fixes coreos/bugs#267 and removes an artificial limitation in systemd.

### CODE REVIEW

File: src/shared/util.c [hunk 1]
```diff
@@ -4280,6 +4280,42 @@ bool hostname_is_valid(const char *s) {
...
+                                return false;
+
+                        dot = false;
+                }
+        }
+
+        if (dot)
+                return false;
+
+
```
Comment (jonboulle) [inline src/shared/util.c]: superfluous newline here

**Comment by marineam** [comment]
I have created a similar function in the past but for a different purpose, that thread may be useful: https://www.mail-archive.com/systemd-devel@lists.freedesktop.org/msg26783.html

At the very least it points to one other place where domain names are validated with the hostname function.

File: src/shared/util.c [hunk 2]
```diff
@@ -4280,6 +4280,41 @@ bool hostname_is_valid(const char *s) {
         return true;
 }
 
+bool domainname_is_valid(const char *s) {
```
Comment (marineam) [inline src/shared/util.c]: This is a big copy/paste, is the only difference between the two the length? If so perhaps use a static function that has a length option to share the implementation goo.

File: src/shared/util.c [hunk 3]
```diff
@@ -4280,6 +4280,41 @@ bool hostname_is_valid(const char *s) {
...
+                                return false;
+
+                        dot = false;
+                }
+        }
+
+        if (dot)
+                return false;
+
+        if (p-s > 255)
```
Comment (marineam) [inline src/shared/util.c]: Should be a symbolic name like a macro

Comment (mischief) [inline src/shared/util.c]: funny that you mention that. where can i find a standardized macro that has the size of a domain as specified in rfc 1035 (255 octets) ?

Comment (marineam) [inline src/shared/util.c]: No idea if it is in any standard headers, at the very least it should be added to a systemd header.

Comment (marineam) [inline src/shared/util.c]: Only things I see related is `arpa/nameser.h` which is specifically for DNS packet stuff:

```c
/*
 * Define constants based on RFC 883, RFC 1034, RFC 1035
 */
#define NS_PACKETSZ     512     /*%< default UDP packet size */
#define NS_MAXDNAME     1025    /*%< maximum domain name */
#define NS_MAXMSG       65535   /*%< maximum message size */
#define NS_MAXCDNAME    255     /*%< maximum compressed domain name */
```

No idea why NS_MAXDNAME is 1025

**Pull request closed without merging**